wcpdf-mpdf.php

```diff
@@ -78,9 +79,41 @@ public function output() {
 endif; // class_exists
 
 add_filter( 'wpo_wcpdf_pdf_maker', 'wpo_wcpdf_pdf_maker_mpdf' );
-function wpo_wcpdf_pdf_maker_mpdf( $class ) {
-	$class = 'WCPDF_Custom_PDF_Maker_mPDF';
-	return $class;
+function wpo_wcpdf_pdf_maker_mpdf( string $class ): string {
+	if ( ! wpo_wcpdf_mpdf_check_dependencies() ) {
+		return $class;
+	}
+
+	return 'WCPDF_Custom_PDF_Maker_mPDF';
+}
+
+add_action( 'admin_init', 'wpo_wcpdf_mpdf_check_dependencies' );
+function wpo_wcpdf_mpdf_check_dependencies(): bool {
+	$activated_plugins       = array_merge( get_option( 'active_plugins', array() ), get_site_option( 'active_sitewide_plugins', array() ) );
+	$core_plugin_slug        = 'woocommerce-pdf-invoices-packing-slips/woocommerce-pdf-invoices-packingslips.php';
+	$core_plugin_min_version = '3.8.4';
+
+	if (
+		! ( in_array( $core_plugin_slug, $activated_plugins ) || array_key_exists( $core_plugin_slug, $activated_plugins ) ) ||
+		! defined( 'WPO_WCPDF_VERSION' ) ||
+		version_compare( WPO_WCPDF_VERSION, $core_plugin_min_version, '<' )
+	) {
+		add_action( 'admin_notices', 'wpo_wcpdf_mpdf_notice_core_plugin_requirement' );
+		return false;
+	}
+
+	return true;
+}
+
+function wpo_wcpdf_mpdf_notice_core_plugin_requirement(): void {
+	/* translators: 1. Plugin name, 2: Core plugin version, 3: Core plugin name, 4: Opening anchor tag, 5: Closing anchor tag */
+	$error_message = sprintf(
+		__( '%1$s requires at least version %2$s of %3$s to be installed and activated.', 'woocommerce-pdf-invoices-packing-slips' ),
+		'<strong>PDF Invoices & Packing Slips for WooCommerce - mPDF</strong>',
+		'<strong>3.8.4</strong>',
```

Comment:
Maybe this can be a variable or a constant?

Reply:
Since defining any variable or constant will be in WordPress's global scope, I was not sure if I should do that. I wanted to prevent adding more stuff to the global scope. Do you think it's OK?

Reply:
Sure, makes sense.